osu.Game/Online/API/APIAccess.cs

```diff
@@ -112,8 +112,14 @@ public APIAccess(OsuGameBase game, OsuConfigManager config, EndpointConfiguratio
 
             config.BindWith(OsuSetting.UserOnlineStatus, configStatus);
 
-            // Early call to ensure the local user / "logged in" state is correct immediately.
-            setPlaceholderLocalUser();
+            if (HasLogin)
+            {
+                // Early call to ensure the local user / "logged in" state is correct immediately.
+                prepareForConnect();
```

Comment:
hmm, i just realise that I renamed this method with the intention of moving the state change into there, but can't as it would lead to it being incorrectly scheduled in the other usage. probably best to revert the name change?

Reply:
(done).